include/boost/geometry/algorithms/detail/distance/segment_to_box.hpp

```diff
@@ -114,7 +115,7 @@ class segment_to_box_2D_generic
         std::vector<box_point> box_points(4);
         detail::assign_box_corners_oriented<true>(box, box_points);
  
-        ps_strategy_type const strategy = strategies.distance(box_points.front(), segment);
+        ps_strategy_type const strategy = strategies.distance(dummy_point(), dummy_segment());
```

Comment:
So we need an instance of a specific type of strategy and therefore we pass the dummies.
Can't we say, for example: `stragies.template distance<point_tag, segment_tag>()` instead? Or are there disadvantages?
What if the returned strategy depends on coordinate system?

Reply:
> Can't we say, for example: `stragies.template distance<point_tag, segment_tag>()` instead?

Yes, we could. At least I think it should be possible to implement this directly inside the umbrella strategies like the current version is. It'd require to modify all of the umbrella strategies.

> Or are there disadvantages?

It's longer and requires the additional `template` keyword. In most of the cases it's enough to simply pass geometry that you already have. So:
```
strategies.distance(point, segment);
strategies.distance(dummy_point(), dummy_segment());
```
vs
```
strategies.template distance<typename tag<point>::type, typename tag<segment>::type>();
strategies.template distance<point_tag, segment_tag>();
```

> What if the returned strategy depends on coordinate system?

The umbrella strategy defines the coordinate system. It's created on the very top of the algorithm. By default it's created based on the coordinate type of the geometries (like it was with legacy strategies). All of the legacy strategies returned by it should use the same coordinate system in order to be consistent with each other. Umbrella strategy is passed internally into all algorithm that may be called by upper-level algorithms. The coordinate systems of geometries are irrelevant. It's because otherwise one could pass upper-level strategy of one CS into an upper-level algorithm which then would call lowel-level algorithm where inconsistent CS would be taken from geometry.

Reply:
Right, thanks for your answers! I'm OK with passing the dummies to avoid longer usage. Usually the geometry will be available, and otherwise, the dummy-definition is quite concise, I noticed. So it's fine from my perspective.